tuned/exports/unix_socket_exporter.py

```diff
@@ -0,0 +1,195 @@
+		self._conn = None
+		self._channel = None
+
+	def running(self):
+		return self._thread is not None
+
+	def export(self, method, in_signature, out_signature):
+		if not ismethod(method):
+			raise Exception("Only bound methods can be exported.")
+
+		method_name = method.__name__
+		if method_name in self._unix_socket_methods:
+			raise Exception("Method with this name (%s) is already exported." % method_name)
+
+		class wrapper(object):
+			def __init__(self, in_signature, out_signature):
+				self._in_signature = in_signature
+				self._out_signature = out_signature
+				
+			def __call__(self, *args, **kwargs):
+				if len(args) + len(kwargs) != len(self._in_signature):
+					# TODO: Add type check for parameters
+					raise Exception("Wrong count of params for method '%s'." % method_name)
+				return method(*args, **kwargs)
+
+		self._unix_socket_methods[method_name] = wrapper(in_signature, out_signature)
+
+	def signal(self, method, out_signature):
+		if not ismethod(method):
+			raise Exception("Only bound methods can be exported.")
+		
+		method_name = method.__name__
+		if method_name in self._unix_socket_methods:
+			raise Exception("Method with this name (%s) is already exported." % method_name)
+		
+		class wrapper(object):
+			def __init__(self, out_signature):
+				self._out_signature = out_signature
+			
+			def __call__(self, *args, **kwargs):
+				return method(*args, **kwargs)
+		
+		self._unix_socket_methods[method_name] = wrapper(out_signature)
+		self._signals.add(method_name)
+
+	def send_signal(self, signal, *args, **kwargs):
+		if not signal in self._signals:
+			raise Exception("Signal '%s' doesn't exist." % signal)
+		for p in self._socket_signal_paths:
+			try:
+				s = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+				s.setblocking(False)
+				s.connect(p)
+				self._send_data(s, "sig", list(args) + list(kwargs.values()))
+				s.close()
+			except OSError as e:
+				log.warning("Error while sending signal '%s' to socket '%s': %s" % (signal, p, e))
+
+	def register_signal_path(self, path):
+		self._socket_signal_paths.append(path)
+
+	def _construct_socket_object(self):
+		if self._socket_path:
+			if os.path.exists(self._socket_path):
+				os.unlink(self._socket_path)
+			self._socket_object = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+			#self._socket_object.setblocking(False)
+			self._socket_object.bind(self._socket_path)
+			self._socket_object.listen(16)
+			os.chown(self._socket_path, self._ownership[0], self._ownership[1])
+			if self._permissions:
+				os.chmod(self._socket_path, self._permissions)
+
+	def start(self):
+		if self.running():
+			return
+
+		self.stop()
+		self._construct_socket_object()
+
+	def stop(self):
+		if self._socket_object:
+			self._socket_object.close()
+
+	def _send_data(self, s, type, data):
+		"""
+		Sends data via socket s.
+		First byte of message specifies type of message (currently "r" for response, "e" for error message and "s" for signal).
```

Comment:
Btw, the code does not match the documentation. You send "err" and "res" here https://github.com/redhat-performance/tuned/pull/470/files#diff-1b8f90567f86039b1dfdcc66a8857985ba22cf3b129e2a8938c8805c6441bf71R191

Reply:
If it would be more comfortable for you, it's small change to pack it to something like 
`{"type": "error", "result": ...}`
It'd would be extendable, if we'd need to change it in future, so I'm pro.
For second one. there's `type[0]` on line 147.